### PR TITLE
feat(worker): convert 'branch' to typescript

### PR DESCRIPTION
**Pull request by viceice**
This pr converts `worker/branch` to typescript

deps: #4433
ref: #1261

### CODE REVIEW

**Comment by rarkins** [comment]
Needs updating from `master`. Doing through the UI isn't working.

**Comment by ViceIce** [comment]
done 😅

**Comment by ViceIce** [comment]
rebased. @JamieMagee can you review?

**Comment by JamieMagee** [comment]
Sure. I don't have time right now, but will look this evening.

File: lib/workers/branch/commit.ts [hunk 1]
```diff
@@ -1,12 +1,22 @@
 import is from '@sindresorhus/is';
+import minimatch from 'minimatch';
 import { platform } from '../../platform';
+import { logger } from '../../logger';
+import { RenovateConfig } from '../../config';
+import { FileData } from '../../platform/common';
```
Comment (JamieMagee) [inline lib/workers/branch/commit.ts]: Can be shortened to 
```suggestion
import { FileData } from '../../platform';
```

File: lib/workers/branch/get-updated.ts [hunk 2]
```diff
@@ -1,17 +1,26 @@
 import is from '@sindresorhus/is';
 import { platform } from '../../platform';
+import { logger } from '../../logger';
+import { get } from '../../manager';
+import { RenovateConfig } from '../../config';
+import { UpdateArtifactsConfig, ArtifactError } from '../../manager/common';
+import { FileData } from '../../platform/common';
```
Comment (JamieMagee) [inline lib/workers/branch/get-updated.ts]: Same here
```suggestion
import { FileData } from '../../platform';
```

**Review by JamieMagee — DISMISSED** [review]

**Comment by ViceIce** [comment]
Thanks @JamieMagee for review, I've merged your changes, have to refactor some more

Last failed build looks like a transient error again 🤔 
https://dev.azure.com/renovatebot/renovate/_build/results?buildId=1155&view=logs

**Comment by ViceIce** [comment]
ok, strange, again a git failure `fatal: not a git repository (or any of the parent directories): .git` 🤔

**Pull request merged**